sphinx_gallery/sorting.py

```diff
@@ -162,7 +162,7 @@ def __call__(self, filename):
         return title
 
 
-class FunctionSortKey:
+def FunctionSortKey(func, r=None):
```

Comment:
opted for CamelCase on the function name, to avoid changing the API (though if you hate this, it's probably no big deal to change it to snake_case, since it wasn't really working before anyway AFAICT).

Reply:
If we allow `within_subsection_order` to take a class instance, then I think this can be reverted and would work on `within_subsection_order` ?

Reply:
yep! done in 716b180a8ab0228b4f807ef2c564b82c3bd9faaf